src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -73,6 +74,97 @@ get_pcidev(const xrt_core::device* device)
   return pcidev::get_dev(device->get_device_id(), device->is_userpf());
 }
 
+static query::sdm_sensor_info::result_type
+get_sdm_sensors(const xrt_core::device* device,
+                const query::sdm_sensor_info::sdr_req_type req_type,
+                const std::string path)
```

Comment:
You forget the references on the last two arguments, don't pass-by-value unless you need the copied value.